crates/sol-macro/src/expand/contract.rs

```diff
@@ -323,24 +323,26 @@ pub(super) fn expand(cx: &ExpCtxt<'_>, contract: &ItemContract) -> Result<TokenS
 
             #[doc = #new_fn_doc]
             #[inline]
-            pub const fn new<P: alloy_contract::private::Provider>(
+            pub const fn new<N: alloy_contract::private::Network, T: alloy_contract::private::Transport + ::core::clone::Clone, P: alloy_contract::private::Provider<N, T>>(
                 address: alloy_sol_types::private::Address,
                 provider: P,
-            ) -> #name<P> {
-                #name::<P>::new(address, provider)
+            ) -> #name<N, T, P> {
+                #name::<N, T, P>::new(address, provider)
             }
 
             #deploy_fn
 
             #[doc = #struct_doc]
             #[derive(Clone)]
-            pub struct #name<P> {
+            pub struct #name<N, T, P> {
                 address: alloy_sol_types::private::Address,
                 provider: P,
+                network: ::core::marker::PhantomData<N>,
+                transport: ::core::marker::PhantomData<T>,
```

Comment:
should be one field and starting with underscore to silence unused warnings

Reply:
got it